### PR TITLE
Add single sign on documentation

### PR DESCRIPTION
**Pull request by martinlagler**
| Q | A
| --- | ---
| Related PRs | sulu/sulu#7262
| License | MIT

#### What's in this PR?

Documentation for Single Sign on configuration.

#### Why?

To document the new feature

### CODE REVIEW

**Comment by Kurt108** [comment]
Is it possible to redirect every user to the IAM-system - no matter which e-mail they are entering for login?

**Comment by alexander-schranz** [comment]
@Kurt108 not yet but think we could add support for something like a `*` domain it should be quiet easy to contribute if you want just something like this and add some tests to it:

Bildschirmfoto 2024-04-16 um 16 46 56

**Comment by Kurt108** [comment]
alright. I will upgrade our current installation and hope our developer can provide the contribution.

File: bundles/security/index.rst [hunk 1]
```diff
@@ -106,6 +106,50 @@ the object type and id are also passed the permissions of the security contexts
 from the role might be overridden by the permissions from this specific object
 (which are handled by the previously mentioned ``AccessControlManager``).
 
+Single-Sign-On  Authentication
```
Comment (Prokyonn) [inline bundles/security/index.rst]:
```suggestion
Single-Sign-On Authentication
```

File: bundles/security/index.rst [hunk 2]
```diff
@@ -106,6 +106,50 @@ the object type and id are also passed the permissions of the security contexts
...
+
+After adjusting the configuration and clearing the symfony cache,
+you only see the ``username or email`` field when you try to login to the administration interface.
+When the user email matches the configured domain,
+the user is then redirected to the SSO provider to authenticate and after successful authentication,
+the user is redirected back to the administration interface.
+If the domain does not match the configured domain, the user is authenticated using the standard login form.
+On password reset, when the domain matches, the user is also redirected to the SSO provider.
+
+Note: At the moment, only the OpenID protocol is supported for Single-Sign-On authentication in Sulu.
```
Comment (Prokyonn) [inline bundles/security/index.rst]:
```suggestion
.. note::

    At the moment, only the OpenID protocol is supported for Single-Sign-On authentication in Sulu.
```

File: bundles/security/index.rst [hunk 3]
```diff
@@ -106,6 +106,50 @@ the object type and id are also passed the permissions of the security contexts
...
+   +        providers:
+   +            'sulu.io':
+   +                dsn: 'openid://%env(resolve:SULU_OPEN_ID_CLIENT_ID)%:%env(resolve:SULU_OPEN_ID_CLIENT_SECRET)%@%env(resolve:SULU_OPEN_ID_ENDPOINT)%'
+   +                user_role: 'USER'
+
+After adjusting the configuration and clearing the symfony cache,
+you only see the ``username or email`` field when you try to login to the administration interface.
+When the user email matches the configured domain,
+the user is then redirected to the SSO provider to authenticate and after successful authentication,
+the user is redirected back to the administration interface.
```
Comment (Prokyonn) [inline bundles/security/index.rst]:
```suggestion
When the user email matches the configured domain,
the user is then redirected to the SSO provider to authenticate. After successful authentication, the system redirects the user back to the administration interface.
```

**Review by Prokyonn — CHANGES_REQUESTED** [review]

File: bundles/security/index.rst [hunk 4]
```diff
@@ -106,6 +106,50 @@ the object type and id are also passed the permissions of the security contexts
...
+    sulu_security:
+        checker:
+            enabled: true
+        password_policy:
+            enabled: true
+   +    single_sign_on:
+   +        providers:
+   +            'sulu.io':
+   +                dsn: 'openid://%env(resolve:SULU_OPEN_ID_CLIENT_ID)%:%env(resolve:SULU_OPEN_ID_CLIENT_SECRET)%@%env(resolve:SULU_OPEN_ID_ENDPOINT)%'
+   +                user_role: 'USER'
```
Comment (alexander-schranz) [inline bundles/security/index.rst]: While write the blog post I'm asking if we should name this `role` or `role_key`. The user role entity is a different one 🤔  /cc @chirimoya @Prokyonn

Comment (chirimoya) [reply bundles/security/index.rst]: `default_role`, `default_role_key`?

Comment (Prokyonn) [reply bundles/security/index.rst]: I think `default_role_key` is the most accurate, I'd go with that

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**